src/promptflow-index/promptflow/index/resources/_index_data_source.py

```diff
@@ -0,0 +1,233 @@
+    :type acs_title_key: str
+    :param acs_metadata_key: The key for the metadata field in the ACS index.
+    :type acs_metadata_key: str
+    :param acs_connection_id: The connection ID for the ACS index.
+    :type acs_connection_id: str
+    :param num_docs_to_import: Number of documents to import from the existing ACS index. Defaults to 50.
+    :type num_docs_to_import: int
+    """
+
+    def __init__(self, *,
+        acs_index_name: str,
+        acs_content_key: str,
+        acs_embedding_key: str,
+        acs_title_key: str,
+        acs_metadata_key: str,
+        acs_connection_id: str,
+        num_docs_to_import: int = 50,
+    ):
+        self.acs_index_name = acs_index_name
+        self.acs_connection_id = acs_connection_id
+        self.acs_content_key = acs_content_key
+        self.acs_embedding_key = acs_embedding_key
+        self.acs_title_key = acs_title_key
+        self.acs_metadata_key= acs_metadata_key
+        self.num_docs_to_import = num_docs_to_import
+        super().__init__(input_type=IndexInputType.AOAI)
+
+    def _createComponent(self, index_config: IndexConfig, acs_config: Optional[AzureAISearchConfig] = None) -> Pipeline:
+        curr_file_path = os.path.dirname(__file__)
+        acs_import_config = json.dumps({"index_name": self.acs_index_name,
+                                        "content_key": self.acs_content_key,
+                                        "embedding_key": self.acs_embedding_key,
+                                        "title_key": self.acs_title_key,
+                                        "metadata_key": self.acs_metadata_key,
+                                        "embedding_model_uri": index_config.embeddings_model,
+                                        })
+        import_acs_component = load_component(os.path.join(curr_file_path, "component-configs", "import_acs_index.yml"))
+
+        rag_job_component: Pipeline =  import_acs_component(
+            embeddings_dataset_name=index_config.output_index_name,
+            embedding_connection=index_config.aoai_connection_id,
+            num_docs_to_import=self.num_docs_to_import,
+            acs_import_connection=self.acs_connection_id,
+            acs_import_config=acs_import_config,
+            data_source_url=index_config.data_source_url
+        )
+        return rag_job_component
+
+class LocalSource(IndexDataSource):
+    """Config class for creating an ML index from a collection of local files.
+
+    :param input_data: An input object describing the local location of index source files.
+    :type input_data: ~azure.ai.ml.Input
+    """
+
+    def __init__(self, *, input_data: str):  # todo Make sure type of input_data is correct
+        self.input_data = Input(type="uri_folder", path=input_data)
+        super().__init__(input_type=IndexInputType.LOCAL)
+
+    def _createComponent(self, index_config: IndexConfig, acs_config: Optional[AzureAISearchConfig] = None) -> Pipeline:
```

Comment:
this function name is not pythonic

Reply:
thanks for catching that! Changed to `_create_component`